src/RadixDlt.NetworkGateway.PostgresIntegration/Services/PendingTransactionResponseAggregator.cs

```diff
@@ -0,0 +1,292 @@
+            GatewayApiSdk.Model.TransactionIntentStatus.LikelyButNotCertainRejection,
+            GatewayApiSdk.Model.TransactionIntentStatus.Pending,
+        };
+
+        var permanentlyRejectsAtEpoch = statusesToReturnPermanentlyRejectsAtEpoch.Contains(intentStatus) ? _rejectionEpoch : null;
+
+        return new GatewayApiSdk.Model.TransactionStatusResponse(
+            ledgerState: _ledgerState,
+            status: legacyIntentStatus,
+            intentStatus,
+            intentStatusDescription: GetIntentStatusDescription(intentStatus),
+            knownPayloads: _knownPayloads,
+            committedStateVersion: _committedStateVersion,
+            permanentlyRejectsAtEpoch: permanentlyRejectsAtEpoch?.ToString(),
+            errorMessage
+        );
+    }
+
+    private string GetPayloadStatusDescription(GatewayApiSdk.Model.TransactionPayloadStatus payloadStatus)
+    {
+        return payloadStatus switch
+        {
+            GatewayApiSdk.Model.TransactionPayloadStatus.Unknown =>
+                "No information is known about the possible outcome of this transaction payload.",
+            GatewayApiSdk.Model.TransactionPayloadStatus.CommittedSuccess =>
+                "This particular payload for this transaction has been committed to the ledger as a success. For more information, use the /transaction/committed-details endpoint.",
+            GatewayApiSdk.Model.TransactionPayloadStatus.CommittedFailure =>
+                "This particular payload for this transaction has been committed to the ledger as a failure. For more information, use the /transaction/committed-details endpoint.",
+            GatewayApiSdk.Model.TransactionPayloadStatus.CommitPendingOutcomeUnknown =>
+                "This particular payload for this transaction has been committed to the ledger, but the Gateway is still waiting for further details about its result.",
+            GatewayApiSdk.Model.TransactionPayloadStatus.PermanentlyRejected =>
+                "This particular payload for this transaction has been permanently rejected. It is not possible for this particular transaction payload to be committed to this network.",
+            GatewayApiSdk.Model.TransactionPayloadStatus.TemporarilyRejected =>
+                "This particular payload for this transaction was rejected at its last execution. It may still be possible for this transaction payload to be committed to this network.",
+            GatewayApiSdk.Model.TransactionPayloadStatus.Pending =>
+                "This particular payload for this transaction has been accepted into a node's mempool on the network. It's possible but not certain that it will be committed to this network.",
+        };
+    }
+
+    private string GetIntentStatusDescription(GatewayApiSdk.Model.TransactionIntentStatus intentStatus)
+    {
+        return intentStatus switch
+        {
+            GatewayApiSdk.Model.TransactionIntentStatus.Unknown =>
```

Comment:
Please use `GatewayModel = RadixDlt.GatewayApiSdk.Model` to keep the codebase consistent.